packages/api/src/controllers/helpers.ts

```diff
@@ -669,34 +669,40 @@ export const triggerCatalystPullStart =
           );
         }
 
-        const deadline = Date.now() + 2 * PULL_START_TIMEOUT;
+        const deadline = Date.now() + PULL_START_TIMEOUT;
         while (Date.now() < deadline) {
-          const res = await fetchWithTimeoutAndRedirects(playbackUrl, {
-            method: "GET",
-            timeout: PULL_START_TIMEOUT,
-            maxRedirects: 10,
-          });
-          const errHeader = res.headers.get("error");
-          const isHlsErr =
-            errHeader &&
-            errHeader != "" &&
-            !errHeader.includes("not allowed to view this stream");
-          if (res.ok && !isHlsErr) {
-            return;
-          }
+          try {
+            const res = await fetchWithTimeoutAndRedirects(playbackUrl, {
+              method: "GET",
+              timeout: PULL_START_TIMEOUT,
+              maxRedirects: 10,
+            });
+            const errHeader = res.headers.get("error");
+            const isHlsErr =
+              errHeader &&
+              errHeader != "" &&
+              !errHeader.includes("not allowed to view this stream");
+            if (res.ok && !isHlsErr) {
+              return;
+            }
 
-          const body = await res.text();
-          logger.warn(
-            `failed to trigger catalyst pull for stream=${
-              stream.id
-            } playbackUrl=${playbackUrl} status=${
-              res.status
-            } error=${JSON.stringify(body)}`,
-          );
+            const body = await res.text();
+            logger.warn(
+              `failed to trigger catalyst pull for stream=${
+                stream.id
+              } playbackUrl=${playbackUrl} status=${
+                res.status
+              } error=${JSON.stringify(body)}`,
+            );
+          } catch (e) {
+            logger.warn(
+              `error while trying to trigger catalyst pull for stream=${
+                stream.id
+              } playbackUrl=${playbackUrl} error=${e}`,
+            );
+          }
           await sleep(250);
         }
-
-        throw new Error(`failed to trigger catalyst pull`);
```

Comment:
So this means that if we can't get the stream pull to start we will return a success for the user as if it worked? Or is this handled somewhere else? Seems "optimistically misleading" to return a success if we didn't get a successful response to the playback URL.

Reply:
Yeah it's not a great solution, I'm just trying to do whatever I can to reduce the response time, if they take over 5 seconds then the customer will time them out and fallback to their provider. But I think maybe I have a better solution now, will open a different PR.